### PR TITLE
feat: Add server option `readOnlyMasterKeyIps` to restrict `readOnlyMasterKey` by IP

### PR DESCRIPTION
**Pull request by mtrezza**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).

## Issue

Add server option `readOnlyMasterKeyIps` to restrict `readOnlyMasterKey` by IP.

### Security Note

This is _not_ a vulnerability fix, as the existing `masterKeyIps` option only restricts the `masterKey` and was never designed to apply to `readOnlyMasterKey`. Each master-level key has its own independent IP restriction option:

- `masterKey` → `masterKeyIps`
- `maintenanceKey` → `maintenanceKeyIps`
- `readOnlyMasterKey` → (no IP restriction option existed)

## Tasks

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, code comments)
- [x] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK 

## Summary by CodeRabbit

* **New Features**
  * IP-based access control for the read-only master key via a new configuration option (defaults allow all).

* **Documentation**
  * Expanded docs detailing formats, CIDR support, defaults, edge cases, and environment-variable usage.

* **Bug Fixes**
  * Improved IPv6 wide-address handling (treats ::0 alongside ::/0 and ::).

* **Tests**
  * Added tests covering allowed and denied IP scenarios for read-only master key usage.

* **Chores / Deprecations**
  * Added a deprecation entry guiding a future default change for the new option.

### CODE REVIEW

**Comment by parseplatformorg** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/acinader/pr-checks/bc204c95-a713-42d9-8fde-b4fa6a7f7139?refs=cd282d59017ab267bfd4dadd6c0fbbced75d4bcf&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: src/Options/index.js [hunk 1]
```diff
@@ -82,6 +82,9 @@ export interface ParseServerOptions {
...
+  :DEFAULT: ["127.0.0.1","::1"] */
+  readOnlyMasterKeyIps: ?(string[]);
```
[the remark this answers is not in the document]
Comment (mtrezza) [reply src/Options/index.js]: Check commit

File: src/Security/CheckGroups/CheckGroupServerConfig.js [hunk 2]
```diff
@@ -117,6 +117,23 @@ class CheckGroupServerConfig extends CheckGroup {
...
+          'The read-only master key can be used from any IP address, which increases the attack surface if the key is compromised.',
+        solution:
+          "Change Parse Server configuration to 'readOnlyMasterKeyIps: [\"127.0.0.1\", \"::1\"]' to restrict access to localhost, or set it to a list of specific IP addresses.",
+        check: () => {
+          if (!config.readOnlyMasterKey) {
+            return;
+          }
+          const ips = config.readOnlyMasterKeyIps || [];
+          const wildcards = ['0.0.0.0/0', '::/0', '::', '::0'];
+          if (ips.some(ip => wildcards.includes(ip))) {
```
[the remark this answers is not in the document]
Comment (mtrezza) [reply src/Security/CheckGroups/CheckGroupServerConfig.js:132]: Check commit

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [9.5.0-alpha.12](https://github.com/parse-community/parse-server/releases/tag/9.5.0-alpha.12)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [9.5.0](https://github.com/parse-community/parse-server/releases/tag/9.5.0)
